src/audio/module_adapter/module/modules.c

```diff
@@ -99,27 +99,24 @@ static int modules_init(struct processing_module *mod)
 		(struct sof_module_api_build_info *)
 		(module_entry->segment[SOF_MAN_SEGMENT_TEXT].v_base_addr);
 
-	void *mod_adp;
-
 	/* Check if module is FDK */
 	if (mod_buildinfo->format == IADK_MODULE_API_BUILD_INFO_FORMAT &&
 	    mod_buildinfo->api_version_number.full == IADK_MODULE_API_CURRENT_VERSION) {
-		mod_adp = system_agent_start(md->module_entry_point, module_id,
-					     instance_id, 0, log_handle, &mod_cfg);
+		md->module_adapter = (void *)system_agent_start(md->module_entry_point, module_id,
+								instance_id, 0, log_handle,
+								&mod_cfg);
 	} else
 	/* Check if module is native */
 	if (mod_buildinfo->format == SOF_MODULE_API_BUILD_INFO_FORMAT &&
 	    mod_buildinfo->api_version_number.full == SOF_MODULE_API_CURRENT_VERSION) {
 		/* If start agent for sof loadable */
 		mod->is_native_sof = true;
-		mod_adp = native_system_agent_start(mod->sys_service, md->module_entry_point,
-						    module_id, instance_id, 0, log_handle,
-						    &mod_cfg);
+		mod->priv.ops = native_system_agent_start(mod->sys_service, md->module_entry_point,
+							  module_id, instance_id, 0, log_handle,
+							  &mod_cfg);
```

Comment:
this is good in principle, but then you need to change all the wrappers in this file doing https://github.com/thesofproject/sof/blob/970188975475d845bb887cce2844c81796a1cec0/src/audio/module_adapter/module/modules.c#L258-L259

Also please use `md` instead of `mod->priv`